bundles/org.openhab.binding.nuki/src/main/java/org/openhab/binding/nuki/internal/discovery/NukiDeviceDiscoveryService.java

```diff
@@ -0,0 +1,105 @@
+/**
+ * Copyright (c) 2010-2021 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.nuki.internal.discovery;
+
+import java.util.Set;
+import java.util.concurrent.ScheduledExecutorService;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.nuki.internal.constants.NukiBindingConstants;
+import org.openhab.binding.nuki.internal.dataexchange.BridgeListResponse;
+import org.openhab.binding.nuki.internal.dto.BridgeApiListDeviceDto;
+import org.openhab.binding.nuki.internal.handler.NukiBridgeHandler;
+import org.openhab.core.common.ThreadPoolManager;
+import org.openhab.core.config.discovery.AbstractDiscoveryService;
+import org.openhab.core.config.discovery.DiscoveryResult;
+import org.openhab.core.config.discovery.DiscoveryResultBuilder;
+import org.openhab.core.thing.ThingUID;
+import org.openhab.core.thing.binding.ThingHandler;
+import org.openhab.core.thing.binding.ThingHandlerService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Discovery service which uses Brige API to find all devices connected to bridges.
+ *
+ * @author Jan Vybíral - Initial contribution
+ */
+@NonNullByDefault
+public class NukiDeviceDiscoveryService extends AbstractDiscoveryService implements ThingHandlerService {
+
+    private final Logger logger = LoggerFactory.getLogger(NukiDeviceDiscoveryService.class);
+    private final ScheduledExecutorService threadPool = ThreadPoolManager.getScheduledPool("discovery");
```

Comment:
What's the reason for creating a custom thread pool field instead of using the provided `scheduler`?